staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_discovery_controller.go

```diff
@@ -96,7 +96,21 @@ func (c *DiscoveryController) sync(version schema.GroupVersion) error {
 			Version:      crd.Spec.Version,
 		})
 
-		if crd.Spec.Version != version.Version {
+		foundThisVersion := false
+		for _, v := range crd.Spec.Versions {
+			if !v.Served {
+				continue
+			}
+			apiVersionsForDiscovery = append(apiVersionsForDiscovery, metav1.GroupVersionForDiscovery{
+				GroupVersion: crd.Spec.Group + "/" + v.Name,
+				Version:      v.Name,
+			})
+			if v.Name == version.Version {
+				foundThisVersion = true
+			}
+		}
+
+		if !foundThisVersion {
 			continue
 		}
 		foundVersion = true
```

Comment:
with multiple versions supported, this statement is no longer true:
> Versions: apiVersionsForDiscovery,
> // the preferred versions for a group is arbitrary since there cannot be duplicate resources
> PreferredVersion: apiVersionsForDiscovery[0],

the ordering of apiVersionsForDiscovery needs to reflect the ordering of the versions in the CRDs for the group (and the [open question](https://github.com/kubernetes/kubernetes/pull/63830/commits/4b38b4966a2c6c74f6c57b4363e15d5f7079267b#r188299377) about conflicting order needs resolving).